openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
+          type: integer
+        - in: query
+          name: gramsYeast
+          required: false
+          type: integer
+          description: Search doughs by grams yeast used
+        - in: query
+          name: gramsSalt
+          required: false
+          type: integer
+          description: Search doughs by grams salt used
+        - in: query
+          name: bulkFermentTime
+          required: false
+          type: integer
+          description: Search doughs by bulk ferment time
+        - in: query
+          name: proofTime
+          required: false
+          type: integer
+          description: Search doughs by proof time
+        - in: query
+          name: waterTemp
+          required: false
+          type: integer
+          description: Search doughs by water temperature at mix time
+        - in: query
+          name: flourType
+          required: false
+          description: Search doughs by flour type
+          type: string
+      responses:
+        200:
+          description: "A list of dough recipes"
+          schema:
+            $ref: "#/definitions/DoughsResult"
+        500:
+          $ref: "#/responses/500"
+    post:
+      description: "Create a dough recipe"
+      tags:
+        - dough
+      parameters:
+        - in: body
+          name: attributes
+          description: attributes of the new dough recipe
+          schema:
+            $ref: "#/definitions/DoughAttributes"
+      responses:
+        200:
+          description: "The created dough recipe"
+          schema:
+            $ref: "#/definitions/DoughResult"
+        500:
+          $ref: "#/responses/500"
+  /doughs/{doughID}:
+    get:
+      description: "Get the dough recipe with ID doughID"
+      tags:
+        - dough
+      parameters:
+        - in: path
+          name: doughID
+          required: true
+          type: string
+          description: ID of the dough recipe to retrieve
+      responses:
+        200:
+          description: "A single dough recipe"
+          schema:
+            $ref: "#/definitions/DoughResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+    patch:
+      description: "Update a dough recipe"
+      tags:
+        - dough
+      parameters:
+        - in: path
+          name: doughID
+          required: true
+          type: string
+          description: ID of the dough recipe to retrieve
+        - in: body
+          name: attributes
+          description: "attributes of the dough recipe to be updated"
+          schema:
+            $ref: "#/definitions/DoughAttributes"
+      responses:
+        200:
+          description: "The updated dough recipe"
+          schema:
+            $ref: "#/definitions/DoughResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+      deprecated: false
+  /doughs/{doughID}/pizzas:
```

Comment:
This endpoint makes me wonder can't we also return dough data when querying `/pizzas`? In other words, what's the main difference between `/doughs/A/pizzas` and `/pizzas?doughID=A`?

Reply:
That sounds like a better approach since it would let you compose that with other filters. I think I'll remove the `/doughs/{doughID}/pizzas` and replace it with the parameter approach.